### ISSUE TITLE
[prefer-regexp-exec] The rule seems to be wrong

### ISSUE THREAD

**Issue by orgads** [typescript-eslint/typescript-eslint#2280]
**Repro**

```JSON
{
  "rules": {
    "@typescript-eslint/prefer-regexp-exec": "error"
  }
}
```

```TS
'foo'.match(/oo/)
```

**Expected Result**
It should not report any error.

**Actual Result**
It claims that `Regexp.exec()` is preferred.

**Additional Info**
The rule was requested in #283.

The rule itself looks wrong. The documentation has a link to [Stack Overflow](https://stackoverflow.com/q/9214754/764870), with a quote I couldn't find in this answer. Actually the word "fast" doesn't appear in this page at all.

[Performance tests](https://jsperf.com/match-vs-exec-1) from [several](https://stackoverflow.com/a/62212132/764870) [other](https://stackoverflow.com/q/15867267/764870) questions don't show any significant improvements for one over the other.

It's unclear where the impression of performance difference came from.

**Versions**

| package                            | version |
| ---------------------------------- | ------- |
| `@typescript-eslint/eslint-plugin` | `3.6.0` |
| `@typescript-eslint/parser`        | `3.6.0` |
| `TypeScript`                       | `3.9.6` |
| `ESLint`                           | `7.4.0` |
| `node`                             | `12.18.0` |
| `npm`                              | `6.14.5` |

**Comment by bradzacher** [comment]
Unsure exactly. People requested the rule, people; people implemented the rule.

If there's now no clear perf benefit as the engines have evolved, then the docs probably need to be updated, though there's nothing "wrong" with the rule.

The rule has always primarily existed to enforce consistency within the codebase, which it still does today.

Happy to accept a PR if you want to update the rules docs.
